constd/main.go

```diff
@@ -104,84 +106,84 @@ func (c *constd) reconcileAtmo(errchan chan error) {
 }
 
 func (c *constd) reconcileConstellation(appSource appsource.AppSource, errchan chan error) {
-	runnables := appSource.Runnables()
+	apps := appSource.Applications()
 
-	for i := range runnables {
-		runnable := runnables[i]
+	for _, app := range apps {
+		runnables := appSource.Runnables(app.Identifier, app.AppVersion)
 
-		if _, exists := c.sats[runnable.FQFN]; !exists {
-			c.sats[runnable.FQFN] = newWatcher(runnable.FQFN, c.logger)
-		}
+		for i := range runnables {
```

Comment:
what's the benefit of this over
```golang
for _, runnable := range runnables {
```
I mean functionally it's identical to
```golang
for i := range runnables {
    runnable := runnables[i]
```
but one line fewer?

Reply:
Since `runnable` is being captured in the `launch` closure, it needs to be stored in a variable that won't be overwritten. If you use the terse `for` syntax, the loop could iterate before the scheduler executes `launch`, causing the wrong pointer to be used. It's a Go quirk that has caught me one too many times :P

Reply:
Oh, TIL! Thank you!